test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs

```diff
@@ -70,6 +70,115 @@ public void TensorFlowTransformMatrixMultiplicationTest()
+                var images = ImageLoaderTransform.Create(env, new ImageLoaderTransform.Arguments()
+                {
+                    Column = new ImageLoaderTransform.Column[1]
+                    {
+                        new ImageLoaderTransform.Column() { Source=  "ImagePath", Name="ImageReal" }
+                    },
+                    ImageFolder = imageFolder
+                }, data);
+                var cropped = ImageResizerTransform.Create(env, new ImageResizerTransform.Arguments()
+                {
+                    Column = new ImageResizerTransform.Column[1]{
+                        new ImageResizerTransform.Column() { Source = "ImageReal", Name= "ImageCropped", ImageHeight =32, ImageWidth = 32, Resizing = ImageResizerTransform.ResizingKind.IsoCrop}
+                    }
+                }, images);
+                var pixels = ImagePixelExtractorTransform.Create(env, new ImagePixelExtractorTransform.Arguments()
+                {
+                    Column = new ImagePixelExtractorTransform.Column[1]{
+                        new ImagePixelExtractorTransform.Column() {  Source= "ImageCropped", Name = "image_tensor", UseAlpha=false, InterleaveArgb=true, Convert = false}
+                    }
+                }, cropped);
+
+                var tf = TensorFlowTransform.Create(env, pixels, model_location,
+                    new[] { "detection_boxes", "detection_scores", "num_detections", "detection_classes" },
+                    new[] { "image_tensor" });
+
+                tf.Schema.TryGetColumnIndex("image_tensor", out int input);
+                tf.Schema.TryGetColumnIndex("detection_boxes", out int boxes);
+                tf.Schema.TryGetColumnIndex("detection_scores", out int scores);
+                tf.Schema.TryGetColumnIndex("num_detections", out int num);
+                tf.Schema.TryGetColumnIndex("detection_classes", out int classes);
+                using (var curs = tf.GetRowCursor(col => col == classes || col == num || col == scores || col == boxes || col == input))
+                {
+                    var getInput = curs.GetGetter<VBuffer<byte>>(input);
+                    var getBoxes = curs.GetGetter<VBuffer<float>>(boxes);
+                    var getScores = curs.GetGetter<VBuffer<float>>(scores);
+                    var getNum = curs.GetGetter<VBuffer<float>>(num);
+                    var getClasses = curs.GetGetter<VBuffer<float>>(classes);
+                    var buffer = default(VBuffer<float>);
+                    var inputBuffer = default(VBuffer<byte>);
+                    while (curs.MoveNext())
+                    {
+                        getInput(ref inputBuffer);
+                        getBoxes(ref buffer);
+                        getScores(ref buffer);
+                        getNum(ref buffer);
+                        getClasses(ref buffer);
+                    }
+                }
```

Comment:
do we want to validate anything here ?

Reply:
The images I am using here don't have any detections, so all of the outputs will be all 0. Once I get the models uploaded, I will also upload some images that have detections, then I can add validation of the outputs.

---
In reply to: [216457130](https://github.com/dotnet/machinelearning/pull/857#discussion_r216457130) [](ancestors = 216457130)